### ISSUE TITLE
Run Functional Test runs slowly compared to MTM Environments

### ISSUE THREAD

**Issue by hqer2k9** [Microsoft/vsts-tasks#2704]
- Running functional tests in TFS 15 with Vnext in comparison to the old system with MTM and Tests Environments, it is awfully slow. it takes like 10 minutes after initial test start, before the first tests are started. And while running the tests, they take longer as normal.
- Distribution of tests is slightly "unhappy", tests get distributed at the beginning of the test run, but if 1 machine is finished, while the other one has still 5 long test runs doesn't make sense. Bucket size was a way more intelligent system because it distributed the tests one after another, with the test distribution now the tests get distributed at the start of the test run. Seems like Bucket size get's ignored at RunSettings file now.

is there are a way to improve this? We have updated to RC2 and we are not happy with the test outcome. Feeling like test tast is a bottleneck [not implying that tfs 15 rc2 is bad, we realy love it! :)]

**Comment by allendm-msft** [comment]
@hqer2k9 
You should try the  "Distribute tests by number of machines" option in the RFT task which mimics earlier functionality. Do note though that even with the earlier approach your problem of execution grouping long running tests together, is not guaranteed to go away since we never considered execution time to determine distribution.

For debugging purposes do share the logs of your build after setting system.debug=true in the build variables.

**Comment by hqer2k9** [comment]
Hey Allen, thanks for the response!

we have split the test run into "short running tests" and "long running tests" to make sure the long running tests are not blocking the other ones. In combination with "Distribute tests by number of machines" it's better now. But one thing is still different: short running tests where running for like 1 to 5 seconds each with lab Environment and MTM, now they take like 20 sec each. Maybe a bug in the test task? Br [I will have to see if im allowed to share logs due to Policy and stuff ;)]

**Comment by allendm-msft** [comment]
@hqer2k9 there should be no change in execution time of the test. There is an overhead of bring-up and tear-down of the machine for the test run but it shouldn't impact the test execution at all. 

Just as a fyi. The difference between the older execution and the new one is that we now use vstest based execution vs the older mstest based execution. Ideally there should be no difference at all. 
If you try to execute your tests from a visual studio command prompt using mstest.exe and then the same tests using vstest.console.exe do you observe the same change in execution times?

If you are still facing issues and have something you can share with us to debug further feel free to reach us at devops _underscore_ tools _at_ microsoft _dot_ com

**Comment by allendm-msft** [comment]
@hqer2k9 let us know if you share more details here or on our official email address.

**Comment by ShakedTBolt** [comment]
Hi,

I'm experiencing the same issue. 2 soap requests tests can take more than 2 minutes, while locally the tests take less than a second.
I'm having some difficulties to change the testsettings to runsettings, even with legacy support.
let me know what info do you need for me if any.

Thanks!

**Comment by allendm-msft** [comment]
We are investigating this again. @SatishThareja will keep you updated.

**Comment by SatishThareja** [comment]
@ShakedTBolt Please share the test execution logs from test agent machine with devops_tools@microsoft.com. You would find logs in 'C:\DtaLogs' folder.

**Comment by ShakedTBolt** [comment]
@SatishThareja  - currently the tests are failing, not sure why. but still, it takes almost 4 minutes to actually fail.

I emailed you the log data.

**Comment by SatishThareja** [comment]
@ShakedTBolt Ack, we have received your email with the logs.

**Comment by ShakedTBolt** [comment]
Hi,

Adding logs from successful running.

Thanks!

Shaked Tuvel
QA Specialist
BOLT Solutions Inc.
Shakedt@boltinc.com<mailto:shakedt@boltinc.com>
www.boltinc.com<http://www.boltinc.com/>

**Comment by ShakedTBolt** [comment]
Ok, i think i found out what the issue is.
I had another issue with the DTAExecutionHost.exe - which crash after restart. I replaced the file with one you gave me (version 14.98.25624.0). once the file was replaced, the crash stop, but i start getting issues with the first step:
DTAExecutionHost, FileMonitor: TryReadToken Waiting for file creation
this step have 3 minutes timeout, and it failed every time to find the file.
So it rather not working or working very slowly.

Any advise?
Thanks!
